src/services/search-service.ts

```diff
@@ -1,18 +1,45 @@
+import { RowDataPacket } from "mysql2";
 import logger from "../utils/logger/logger";
-
-// import SQLService from "./sql-service";
+import SQLService from "./sql-service";
 
 class SearchService {
-	static async search() {
-		logger.info("Search");
+	static async search(
+		table: string,
+		key: string,
+		columns: string[]
+	): Promise<RowDataPacket[]> {
+		logger.info(
+			`SearchService invoked! Table = ${table}, Key = ${key}, Columns = ${columns}`
+		);
+
+		const whereParams = ` WHERE MATCH (${columns})`;
+		const againstParams = ` AGAINST ('${key}' IN NATURAL LANGUAGE MODE)`;
+
+		const SQL = `SELECT * FROM ${table}${whereParams}${againstParams};`;
+
+		const rows = await SQLService.query(SQL);
+		return rows;
 	}
 
-	static async searchContacts() {
-		logger.info("Search Contacts");
+	static async searchContacts(key: string): Promise<RowDataPacket[]> {
+		logger.info(`SearchService.searchContacts invoked! Key = ${key}`);
+
+		const rows = SearchService.search("contact", key, [
+			"contact_id",
+			"uh_id",
+			"email",
+			"first_name",
+			"last_name",
+			"phone_number",
+		]);
+		return rows;
 	}
 
-	static async searchEvents() {
-		logger.info("Search Events");
+	static async searchEvents(key: string): Promise<RowDataPacket[]> {
+		logger.info(`SearchService.searchEvents invoked! Key = ${key}`);
+
+		const rows = SearchService.search("event", key, ["title", "description"]);
```

Comment:
What was the result on being able to search DATE columns?

Reply:
MySQL fulltext indexes doesn't support `datetime` columns. They only support `CHAR, VARCHAR, TEXT` columns. I implemented the indexes for any section possible aside from Contact table's `shirt_size_id`.